shell/platform/android/vsync_waiter_android.cc

```diff
@@ -66,8 +90,35 @@ void VsyncWaiterAndroid::OnVsyncFromNDK(int64_t frame_nanos, void* data) {
                    "frame_target_time",
                    target_time.ToEpochDelta().ToMicroseconds());
 
-  auto* weak_this = reinterpret_cast<std::weak_ptr<VsyncWaiter>*>(data);
-  ConsumePendingCallback(weak_this, frame_time, target_time);
+  auto* weak_this = reinterpret_cast<std::weak_ptr<VsyncWaiterAndroid>*>(data);
+  ConsumePendingCallback(weak_this, frame_time, target_time, 0);
+}
+
+// static
+void VsyncWaiterAndroid::OnVsyncFromNDK33(
+    AndroidChoreographer::AChoreographerFrameCallbackData* callback_data,
+    void* data) {
+  auto frame_time = AndroidChoreographer::GetFrameTime(callback_data);
+  auto now = fml::TimePoint::Now();
+  if (frame_time > now) {
+    frame_time = now;
+  }
+  auto* weak_this = reinterpret_cast<std::weak_ptr<VsyncWaiterAndroid>*>(data);
+  auto shared_this = weak_this->lock();
+  if (!shared_this) {
+    delete weak_this;
```

Comment:
It'd be nice if we could avoid this in the NDK based code paths by making `ConsumePendingCallback` not do manual deletion and instead just do it in the JDK based implementation, but that can be done separately from this PR.